cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -367,12 +370,8 @@ if (ARROW_IPC)
     set(FLATBUFFERS_VENDORED 0)
   endif()
 
-  if(RAPIDJSON_VENDORED)
-    set(ARROW_DEPENDENCIES ${ARROW_DEPENDENCIES} rapidjson_ep)
-  endif()
-
   if(FLATBUFFERS_VENDORED)
-    set(ARROW_DEPENDENCIES ${ARROW_DEPENDENCIES} flatbuffers_ep)
```

Comment:
For future reference, this way of declaring dependencies as a list with EP's seems to not work with all versions of CMake. It works with library targets though for some reason